pkg/kubevirt/core/core.go

```diff
@@ -209,10 +198,7 @@ func (p PluginSPIImpl) CreateMachine(ctx context.Context, machineName string, pr
 							},
 							Interfaces: interfaces,
 						},
-						Resources: kubevirtv1.ResourceRequirements{
-							Requests: *requestsAndLimits,
-							Limits:   *requestsAndLimits,
-						},
+						Resources: *providerSpec.Resources,
```

Comment:
Potential nil pointer panic: `providerSpec.resources` could be nil. Can be avoided by using a non-pointer type.